src/modules/pages/SelectPhotos.tsx

```diff
@@ -0,0 +1,131 @@
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { Button, IconButton, ImageList, ImageListItem, Typography, Checkbox, ImageListItemBar } from '@mui/material';
+import { Box } from '@mui/system';
+import PageTemplate from 'modules/components/PageTemplate';
+import { useNavigate, useParams } from 'react-router-dom';
+import CloseIcon from '@mui/icons-material/Close';
+import { useAppDispatch } from 'shared/hooks/redux-hooks';
+import { colors } from 'styles/variables';
+import { ChangeEvent, useEffect, useState } from 'react';
+import { IFABProp } from 'shared/interfaces/selectPhotos.interface';
+import { getBase64StringFromDataURL, toBase64 } from 'shared/helpers/toolsBase64';
+// import API from 'core/services/API';
+import { IPhoto } from 'shared/interfaces/photo.interface';
+import { createPhoto, retrievePhotos } from 'shared/store/Photos/photoSlice';
+import Endpoints from 'shared/constants/endpoints';
+import { retrieveAlbum, updateAlbum } from 'shared/store/Album/albumSlice';
+import { IAlbum } from 'shared/interfaces/album.interface';
+
+const SelectPhotos = (): JSX.Element => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [selectedPhotoId, setSelectedPhotoId] = useState<number[]>([]);
+  const [album, setAlbum] = useState<IAlbum>();
+  const [allPhotos, setAllPhotos] = useState<IPhoto[]>([]);
+
+  const handleClickDone = () => {
+    if (album) {
+      // add photo in album and check if photo exist in album
+      dispatch(updateAlbum({
+        ...album,
+        photos: [...album.photos, ...selectedPhotoId.filter((item) => !album.photos.includes(item))],
+      })).unwrap().then(() => {
+        navigate(`${Endpoints.Album}/${id}`);
+      });
+    }
+  };
+
+  const onImageChange = (ev: ChangeEvent<HTMLInputElement>) => {
+    if (ev.target.files) {
+      const arr = Array.from(ev.target.files);
+      const p = arr.map(async (item) => {
+        const result = await toBase64(item);
+        const base64 = getBase64StringFromDataURL(result as string);
+        const newPhoto: Omit<IPhoto, 'id'> = {
+          date: item.lastModified,
+          description: 'def',
+          image: base64,
+          size: item.size,
+          type: item.type,
+        };
+        return dispatch(createPhoto(newPhoto)).unwrap();
+      });
+
+      Promise.all(p).then(response => {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled try this one instead

Reply:
+, refactored this part  in branch feature/view-single-photo-hw2 in commit [612ae69]